kong/router/atc.lua

```diff
@@ -522,10 +595,26 @@ do
         value = value:lower()
       end
 
-      headers_buf:putf("|%s=%s", name, value)
+      str_buf:putf("|%s=%s", name, value)
     end
 
-    return headers_buf:get()
+    return str_buf:get()
+  end
+
+  get_queries_key = function(queries)
```

Comment:
I think this function can be reused with `get_headers_key`:

What about: `gen_cache_key_from_table = function(tbl, header)`

Where if `header = true`,  `local name = name:gsub("-", "_"):lower()` will be executed.

Reply:
Okay, we can leave this as-is for now.